ecs-cli/modules/utils/compose/ecs_params_reader.go

```diff
@@ -52,6 +56,23 @@ type ContainerDef struct {
 	Cpu               int64                  `yaml:"cpu_shares"`
 	Memory            libYaml.MemStringorInt `yaml:"mem_limit"`
 	MemoryReservation libYaml.MemStringorInt `yaml:"mem_reservation"`
+	HealthCheck       HealthCheck            `yaml:"healthcheck"`
+}
+
+// HealthCheck holds the ECS container health check
+// https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_HealthCheck.html
+type HealthCheck struct {
```

Comment:
So the reason why I did this is because the custom unmarshal function needs to be on a type. I didn't want to put the custom unmarshal function on a type (ecs.HealthCheck) that was from the SDK, I thought it should be on a struct which we create.

Reply:
Hmm, at that point could you not just alias rather than embed the type? e.g. `type HeathCheck ecs.HealthCheck`